lua/neotest-vitest/init.lua

```diff
@@ -64,22 +64,25 @@ function adapter.is_test_file(file_path)
   if file_path == nil then
     return false
   end
-
-  if not hasVitestDependency(file_path) then
-    return false
-  end
+  local is_test_file = false
+  local is_test_folder = false
 
   if string.match(file_path, "__tests__") then
-    return true
+    is_test_folder = true
   end
 
   for _, x in ipairs({ "spec", "test" }) do
     for _, ext in ipairs({ "js", "jsx", "coffee", "ts", "tsx" }) do
       if string.match(file_path, "%." .. x .. "%." .. ext .. "$") then
-        return true
+        is_test_file = true
+        goto matched_pattern
       end
     end
   end
+  ::matched_pattern::
+  if (is_test_file or is_test_folder) and hasVitestDependency(file_path) then
+    return true
+  end
 
   return false
```

Comment:
I think we can simplify in this way:
```suggestion
  return is_test_file and hasVitestDependency(file_path)
```